sdk/python/feast/infra/online_stores/dynamodb.py

```diff
@@ -207,28 +208,59 @@ def online_read(
             config: The RepoConfig for the current FeatureStore.
             table: Feast FeatureView.
             entity_keys: a list of entity keys that should be read from the FeatureStore.
+            sort_response: wether or not to sort DynamoDB responses by the entity_ids order.
         """
         online_config = config.online_store
         assert isinstance(online_config, DynamoDBOnlineStoreConfig)
         dynamodb_resource = self._get_dynamodb_resource(online_config.region)
+        table_instance = dynamodb_resource.Table(
+            _get_table_name(online_config, config, table)
+        )
 
         result: List[Tuple[Optional[datetime], Optional[Dict[str, ValueProto]]]] = []
-        for entity_key in entity_keys:
-            table_instance = dynamodb_resource.Table(
-                _get_table_name(online_config, config, table)
-            )
-            entity_id = compute_entity_id(entity_key)
+        entity_ids = [compute_entity_id(entity_key) for entity_key in entity_keys]
+
+        len_entity_ids = len(entity_ids)
+        batch_size = 40
+        # Iterate until the end_index is the value len_entity_ids
+        iters = (
+            len_entity_ids // batch_size + 1
+            if len_entity_ids % batch_size > 0
+            else len_entity_ids // batch_size
+        )
+        for i in range(iters):
+            start_index = min(i * batch_size, len_entity_ids)
+            end_index = min(i * batch_size + batch_size, len_entity_ids)
+
+            batch_entity_ids = {
+                table_instance.name: {
+                    "Keys": [
+                        {"entity_id": entity_id}
+                        for entity_id in entity_ids[start_index:end_index]
```

Comment:
seems like this might be off by one? Would capture up to `batch_size - 1` in each iteration i think since slices are upper bound exclusive.